File: Fortran/011.f90
program leetcode011
    implicit none
    integer(kind=4),parameter :: arr_size = 9
    integer(kind=4),dimension(arr_size) :: arr = (/1,8,6,2,5,4,8,3,7/)
    integer(kind=4) :: area = 0
    call maxArea(arr,arr_size,area)
    write(*,*) 'The max area is', area
end program leetcode011

subroutine maxArea(arr,arr_size,area)
    implicit none
    integer(kind=4),intent(in) :: arr_size
    integer(kind=4),dimension(arr_size) :: arr
    integer(kind=4) :: area
    integer(kind=4) :: i,j
    i=1
    j=arr_size
    do while(i<j)
        area = max(area,min(arr(i),arr(j))*(j-i))
        if(arr(i)<arr(j))then
            i=i+1
        else
            j=j-1
        end if
    end do
    return
end subroutine maxArea
!gfortran 011.f90
!./a.out
!output 49

File: Python/011.py
class Solution011(object):
    def maxArea(self, height):
        i = 0 #最左边
        j = len(height) - 1 #最右边
        maxval = 0
        while i < j:
            maxval = max(maxval, min(height[i], height[j])*(j-i)) #找出最左边和最右边组成容器的容积
            if height[i] < height[j]: #短边不需要再考虑 因为此时只要选定短边 其他任选边的容积都不会大于maxval
                i += 1 #短边不用考虑 考虑下一块木板
            else:
                j -= 1 #短边不用考虑 考虑下一块木板
        return maxval
test011=Solution011()
nums011=[1,8,6,2,5,4,8,3,7]
test011.maxArea(nums011)
#output 49
